main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -1252,11 +1248,7 @@ object Resolver extends Phase[NamedAst.Root, ResolvedAst.Root] {
           case (None, Some(typealias)) => getTypeAliasIfAccessible(typealias, ns0, root, loc)
 
           // Case 4: Errors.
-          case (x, y) =>
-            val loc1 = x.map(_.loc)
-            val loc2 = y.map(_.loc)
-            val locs = List(loc1, loc2).flatten
-            ResolutionError.AmbiguousType(typeName, ns0, locs, loc).toFailure
+          case (x, y) => throw InternalCompilerException("Unexpected ambiguous type.")
```

Comment:
Is this because we assume such ambiguity to have already been resolved at this point? Maybe we should write:

"Unexpected ambiguity: Duplicate types / classes should have been resolved."